source/common/event/dispatcher_impl.cc

```diff
@@ -162,14 +162,13 @@ void DispatcherImpl::run(RunType type) {
 }
 
 void DispatcherImpl::runPostCallbacks() {
-  std::unique_lock<std::mutex> lock(post_lock_);
-  while (!post_callbacks_.empty()) {
-    std::function<void()> callback = post_callbacks_.front();
+  for (std::function<void()> callback; true; callback()) {
```

Comment:
This is an unconventional style of for loop. Can you write this as "while (true)" and put callback() in the body please?

Reply:
heh OK....thought it was worth a taste test.  That's the previous commit: https://github.com/envoyproxy/envoy/pull/3427/commits/7fed761ce0a11a9d7b49295f447e65f3f27b107d

will revert to that.